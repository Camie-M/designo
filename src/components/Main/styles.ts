'use client'

import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: ${(props) => props.theme.palette.light};
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  h1,
  h2,
  p {
    color: ${(props) => props.theme.palette.darkest};
  }

  h1 {
    font-size: ${(props) => props.theme.fonts.h1.fontSize};
    font-weight: ${(props) => props.theme.fonts.h1.fontWeight};
    line-height: ${(props) => props.theme.fonts.h1.lineHeight};
  }

  h2 {
    font-size: ${(props) => props.theme.fonts.h2.fontSize};
    font-weight: ${(props) => props.theme.fonts.h2.fontWeight};
    line-height: ${(props) => props.theme.fonts.h2.lineHeight};
    letter-spacing: ${(props) => props.theme.fonts.h2.letterSpacing};
  }

  h3 {
    color: ${(props) => props.theme.palette.primary};
    font-size: ${(props) => props.theme.fonts.h3.fontSize};
    font-weight: ${(props) => props.theme.fonts.h3.fontWeight};
    line-height: ${(props) => props.theme.fonts.h3.lineHeight};
    letter-spacing: ${(props) => props.theme.fonts.h3.letterSpacing};
    text-transform: ${(props) => props.theme.fonts.h3.textTransform};
  }

  p {
    font-size: ${(props) => props.theme.fonts.text.fontSize};
    font-weight: ${(props) => props.theme.fonts.text.fontWeight};
    line-height: ${(props) => props.theme.fonts.text.lineHeight};
  }
`
