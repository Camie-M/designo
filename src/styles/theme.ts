'use client'

import { DefaultTheme } from 'styled-components'

export const defaultTheme: DefaultTheme = {
  borderRadius: '8px',
  palette: {
    primary: '#e7806b',
    primaryLight: '#ffac9b',
    font: '#1d1c1e',
    fontLight: '#333136',
    contrastMain: '#fff',
    contrastSecondary: '#f3f5f6'
  },
  fonts: {
    h1: {
      fontSize: '48px',
      fontWeight: 500,
      lineHeight: '48px'
    },
    h2: {
      fontSize: '40px',
      fontWeight: 500,
      lineHeight: '48px',
      letterSpacing: '2px'
    },
    h3: {
      fontSize: '20px',
      fontWeight: 500,
      lineHeight: '26px',
      letterSpacing: '5px',
      textTransform: 'uppercase'
    },
    text: {
      fontSize: '16px',
      fontWeight: 400,
      lineHeight: '26px'
    }
  }
}
