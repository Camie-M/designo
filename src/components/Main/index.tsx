import Image from 'next/image'

import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <h1>Header</h1>
    <h2>Subtitle</h2>
    <h3>Title</h3>
    <p>Regular text</p>

    <Image src={`/assets/circle.png`} alt={'Circle'} width={640} height={640} />
  </S.Wrapper>
)

export default Main
