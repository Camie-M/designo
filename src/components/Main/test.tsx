import { render, screen } from '@testing-library/react'
import { defaultTheme } from '../../styles/theme'
import { ThemeProvider } from 'styled-components'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Main />
      </ThemeProvider>
    )
    expect(screen.getByRole('heading', { name: /Header/i })).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render the colors correctly', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Main />
      </ThemeProvider>
    )

    expect(container.firstChild).toHaveStyle({
      'background-color': defaultTheme.palette.contrastSecondary
    })
  })
})
