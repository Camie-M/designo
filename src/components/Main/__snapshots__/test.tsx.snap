// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #f3f5f6;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.c0 h1,
.c0 h2,
.c0 p {
  color: #1d1c1e;
}

.c0 h1 {
  font-size: 48px;
  font-weight: 500;
  line-height: 48px;
}

.c0 h2 {
  font-size: 40px;
  font-weight: 500;
  line-height: 48px;
  letter-spacing: 2px;
}

.c0 h3 {
  color: #e7806b;
  font-size: 20px;
  font-weight: 500;
  line-height: 26px;
  letter-spacing: 5px;
  text-transform: uppercase;
}

.c0 p {
  font-size: 16px;
  font-weight: 400;
  line-height: 26px;
}

<main
  class="c0"
>
  <h1>
    Header
  </h1>
  <h2>
    Subtitle
  </h2>
  <h3>
    Title
  </h3>
  <p>
    Regular text
  </p>
  <img
    alt="Circle"
    data-nimg="1"
    decoding="async"
    height="640"
    loading="lazy"
    src="/_next/image?url=%2Fassets%2Fcircle.png&w=1920&q=75"
    srcset="/_next/image?url=%2Fassets%2Fcircle.png&w=640&q=75 1x, /_next/image?url=%2Fassets%2Fcircle.png&w=1920&q=75 2x"
    style="color: transparent;"
    width="640"
  />
</main>
`;
